### ISSUE TITLE
[Bug] BetterX & Eden Ring Crash with Moonlight Library

### ISSUE THREAD

**Issue by TrueDevron** [quiqueck/BetterNether#61]
### What happened?

Have not tried to reproduce on a fabric client. But currently BetterX mods & Eden Ring both crash when installed with Moonlight (Selene) Library on Quilt.

Have ensured each mod is on the latest version but the crash still occurs.

### BetterNether

7.1.0

### BCLib

2.1.0

### Fabric API

0.61.0

### Fabric Loader

0.17.5-beta2

### Minecraft

1.19.1

### Relevant log output

```shell
14:13:11] [main/WARN]: @Final field field_6247:Lnet/minecraft/class_2940; in amethyst_imbuement.mixins.json:LivingEntityMixin from mod amethyst_imbuement should be final
[14:13:11] [main/WARN]: Discarding @Unique public method getAnimatedSkin in 3dskinlayers.mixins.json:DonorAbstractClientPlayerMixin from mod skinlayers because it already exists in net.minecraft.class_742
[14:13:11] [main/WARN]: Discarding @Unique public method getAnimatedSkin in waveycapes.mixins.json:DonorAbstractClientPlayerMixin from mod waveycapes because it already exists in net.minecraft.class_742
[14:13:11] [main/WARN]: Method overwrite conflict for bounce in amethyst_imbuement.mixins.json:BlockMixin from mod amethyst_imbuement, previously written by org.cloudwarp.doodads.mixin.BlockMixin. Skipping method.
[14:13:12] [main/INFO]: [bclib] Added Config bclib.generator to auto sync (file hash)
[14:13:12] [main/INFO]: [bclib] Added Config bclib.main to auto sync (content diff)
[14:13:12] [main/INFO]: [bclib] Added Config bclib.recipes to auto sync (file hash)
[14:13:12] [main/INFO]: [bclib] Added Config bclib.biomes to auto sync (file hash)
[14:13:12] [main/WARN]: Method overwrite conflict for method_32346 in inspecio.mixins.json:PotionItemMixin from mod inspecio, previously written by net.dehydration.mixin.PotionItemMixin. Skipping method.
[14:13:14] [main/INFO]: Building optimized datafixer
[14:13:17] [main/INFO]: [bclib] Make sure features are loaded from terrablender
BCLib - Nether BiomeSource (487f4508)
    biomes     = 5
    namespaces = minecraft
    seed       = 0
    height     = 128
    config     = BCLibNetherBiomeSourceConfig{mapVersion=hex}
[14:13:17] [main/INFO]: [bclib] Make sure features are loaded from terrablender
BCLib - The End BiomeSource (7f02462a)
    biomes     = 5
    namespaces = minecraft
    seed       = 0
    height     = 128
    deciders   = 0
    config     = BCLEndBiomeSourceConfig{mapVersion=hex, generatorVersion=paulevs, withVoidBiomes=true, innerVoidRadiusSquared=1048576, voidBiomesSize=256, centerBiomesSize=256, landBiomesSize=256, barrensBiomesSize=256}
[14:13:17] [main/INFO]: [bclib] Make sure features are loaded from terrablender
BCLib - Nether BiomeSource (7669a108)
    biomes     = 5
    namespaces = minecraft
    seed       = 0
    height     = 128
    config     = BCLibNetherBiomeSourceConfig{mapVersion=square}
[14:13:17] [main/INFO]: [bclib] Make sure features are loaded from terrablender
BCLib - The End BiomeSource (6a98e073)
    biomes     = 5
    namespaces = minecraft
    seed       = 0
    height     = 128
    deciders   = 0
    config     = BCLEndBiomeSourceConfig{mapVersion=square, generatorVersion=paulevs, withVoidBiomes=true, innerVoidRadiusSquared=1048576, voidBiomesSize=256, centerBiomesSize=256, landBiomesSize=256, barrensBiomesSize=256}
[14:13:17] [main/ERROR]: Registry 'enhancedcelestials:entity_filter' was empty after loading
[14:13:19] [main/INFO]: [Animal Feeding Trough] Load Complete! Enjoy :D
[14:13:19] [main/WARN]: Static binding violation: PRIVATE @Overwrite method method_23182 in sodium.mixins.json:features.item.MixinItemRenderer from mod sodium cannot reduce visibiliy of PUBLIC target method, visibility will be upgraded.
[14:13:20] [main/INFO]: [Quilt Command] Networking support is enabled
[14:13:20] [main/INFO]: [bclib] Added Config betternether.main to auto sync (file hash)
[14:13:20] [main/INFO]: [bclib] Added Config betternether.generator to auto sync (file hash)
[14:13:20] [main/INFO]: [bclib] Added Config betternether.blocks to auto sync (file hash)
[14:13:20] [main/INFO]: [bclib] Added Config betternether.items to auto sync (file hash)
[14:13:20] [main/INFO]: [bclib] Added Config betternether.biomes to auto sync (file hash)
[14:13:20] [main/INFO]: [bclib] Added Config betternether.mobs to auto sync (file hash)
[14:13:20] [main/INFO]: [bclib] Added Config betternether.recipes to auto sync (file hash)
[14:13:20] [main/WARN]: An unregistered entry on the Registry[ResourceKey[minecraft:root / minecraft:block] (Experimental)] registry was attempted to be registered in a content registry through the Quilted Fabric API bridge! Its instability will be circumvented through the activation of the registry's deferring queue, which may affect performance!
[14:13:20] [main/ERROR]: Minecraft has crashed!
org.quiltmc.loader.impl.FormattedException: java.lang.ExceptionInInitializerError
	at org.quiltmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:425) ~[quilt-loader-0.17.5-beta.2.jar:?]
	at org.quiltmc.loader.impl.launch.knot.Knot.launch(Knot.java:76) [quilt-loader-0.17.5-beta.2.jar:?]
	at org.quiltmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:24) [quilt-loader-0.17.5-beta.2.jar:?]
Caused by: java.lang.ExceptionInInitializerError
	at org.betterx.betternether.registry.NetherBlocks.<clinit>(NetherBlocks.java:762) ~[better-nether-7.1.0.jar:?]
	at org.betterx.betternether.entity.EntityFirefly.<clinit>(EntityFirefly.java:549) ~[better-nether-7.1.0.jar:?]
	at org.betterx.betternether.registry.NetherEntities.<clinit>(NetherEntities.java:152) ~[better-nether-7.1.0.jar:?]
	at net.minecraft.class_5135.handler$bnb000$hasDefinition(class_5135.java:1031) ~[client-intermediary.jar:?]
	at net.minecraft.class_5135.method_26875(class_5135.java) ~[client-intermediary.jar:?]
	at net.minecraft.class_5135.method_26876(class_5135.java:176) ~[client-intermediary.jar:?]
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:178) ~[?:?]
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179) ~[?:?]
	at java.util.Iterator.forEachRemaining(Iterator.java:133) ~[?:?]
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596) ~[?:?]
	at net.minecraft.class_5135.method_26872(class_5135.java:178) ~[client-intermediary.jar:?]
	at net.minecraft.class_2966.method_17598(class_2966.java:136) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.method_44604(Main.java:162) ~[quilt-loader-0.17.5-beta.2-1.19.2.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:51) ~[quilt-loader-0.17.5-beta.2-1.19.2.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at org.quiltmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:423) ~[quilt-loader-0.17.5-beta.2.jar:?]
	... 2 more
Caused by: java.lang.IllegalStateException: Feature configuration for betternether:temp_patch_twisting_vine can not be null!
	at org.betterx.bclib.api.v3.levelgen.features.BCLFeatureBuilder.buildAndRegister(BCLFeatureBuilder.java:226) ~[bclib-2.1.0.jar:?]
	at org.betterx.bclib.api.v3.levelgen.features.BCLFeatureBuilder.build(BCLFeatureBuilder.java:238) ~[bclib-2.1.0.jar:?]
	at org.betterx.bclib.api.v3.levelgen.features.BCLFeatureBuilder.inlinePlace(BCLFeatureBuilder.java:242) ~[bclib-2.1.0.jar:?]
	at org.betterx.betternether.registry.features.configured.NetherVines.<clinit>(NetherVines.java:113) ~[better-nether-7.1.0.jar:?]
	at org.betterx.betternether.registry.NetherBlocks.<clinit>(NetherBlocks.java:762) ~[better-nether-7.1.0.jar:?]
	at org.betterx.betternether.entity.EntityFirefly.<clinit>(EntityFirefly.java:549) ~[better-nether-7.1.0.jar:?]
	at org.betterx.betternether.registry.NetherEntities.<clinit>(NetherEntities.java:152) ~[better-nether-7.1.0.jar:?]
	at net.minecraft.class_5135.handler$bnb000$hasDefinition(class_5135.java:1031) ~[client-intermediary.jar:?]
	at net.minecraft.class_5135.method_26875(class_5135.java) ~[client-intermediary.jar:?]
	at net.minecraft.class_5135.method_26876(class_5135.java:176) ~[client-intermediary.jar:?]
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:178) ~[?:?]
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179) ~[?:?]
	at java.util.Iterator.forEachRemaining(Iterator.java:133) ~[?:?]
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596) ~[?:?]
	at net.minecraft.class_5135.method_26872(class_5135.java:178) ~[client-intermediary.jar:?]
	at net.minecraft.class_2966.method_17598(class_2966.java:136) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.method_44604(Main.java:162) ~[quilt-loader-0.17.5-beta.2-1.19.2.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:51) ~[quilt-loader-0.17.5-beta.2-1.19.2.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at org.quiltmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:423) ~[quilt-loader-0.17.5-beta.2.jar:?]
	... 2 more
```

**Comment by TrueDevron** [comment]
Forgot to mention that there was no crash report left. Only the latest.log showed what happened

**Comment by quiqueck** [comment]
Might be a duplicate of #48

**Comment by quiqueck** [comment]
Should be fixed in BCLib 2.1.1. Please re-open this issue if you still experience this problem.
